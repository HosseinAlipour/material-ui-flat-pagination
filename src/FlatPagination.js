'use strict';

import PropTypes from 'prop-types';
import React from 'react';
import FlatButton from 'material-ui/FlatButton';

const _ellipsisLabel = '...';

const _styles = {
  buttonStyle: {
    minWidth: 'none'
  },
  labelStyle: {
    padding: '0 12px'
  },
  ellipsisLabelStyle: {
    padding: '0 4px'
  }
};

const validateNumber = (min) => {
  return (props, propName, componentName) => {
    const value = props[propName];
    const type = typeof value;
    if (type !== 'number') {
      return new Error(
        'Failed prop type: Invalid prop `' + propName + '` of type `' + type +
        '` supplied to `' + componentName + '`, expected `number`.'
      );
    }
    const intValue = parseInt(value, 10);
    if (intValue < min) {
      return new Error(
        'Failed prop value: Invalid prop `' + propName + '` of value `' + value +
        '` supplied to `' + componentName + '`, expected a number greater than or equal to `' + min + '`.'
      );
    }
  };
};

class FlatPagination extends React.PureComponent {

  static propTypes = {
    offset: validateNumber(0),
    limit: validateNumber(1),
    total: validateNumber(0),
    className: PropTypes.string,
    currentPageLabelStyle: PropTypes.object,
    currentPageStyle: PropTypes.object,
    disabled: PropTypes.bool,
    disableTouchRipple: PropTypes.bool,
    hoverColor: PropTypes.string,
    nextPageLabel: PropTypes.node,
    onClick: PropTypes.func,
    otherPageLabelStyle: PropTypes.object,
    otherPageStyle: PropTypes.object,
    previousPageLabel: PropTypes.node,
    rippleColor: PropTypes.string,
    reduced: PropTypes.bool,
    style: PropTypes.object
  };

  static defaultProps = {
    disabled: false,
    disableTouchRipple: false,
    nextPageLabel: '>',
    previousPageLabel: '<',
    reduced: false
  };

  static createLabel(offset, limit, total, reduced) {
    const currentPage = Math.floor(offset / limit) + 1;
    const minPage = 1;
    const maxPage = Math.floor(total / limit) + (total % limit === 0 ? 0 : 1);

    const reserveBaseCount = reduced ? 1 : 2;
    const getReserveCount = (anotherSidePageCount) => {
      let ret = reserveBaseCount * 2 - anotherSidePageCount;
      return ret < reserveBaseCount ? reserveBaseCount : ret;
    };

    const leftLabels = (() => {
      const labels = [];
      const reserveCount = getReserveCount(maxPage - currentPage);
      const endReservePage = minPage + (reserveBaseCount - 1);
      const endReserveBoundaryPage = endReservePage + 1;
      let ellipsis = false;
      for (let i = currentPage - 1; i >= minPage; i--) {
        if (labels.length < reserveCount
          || i <= endReservePage
          || (i === endReserveBoundaryPage && !ellipsis)) {
          labels.push('' + i);
        } else if (!ellipsis) {
          ellipsis = true;
          labels.push(_ellipsisLabel);
        }
      }
      return labels;
    })();

    const rightLabels = (() => {
      const labels = [];
      const reserveCount = getReserveCount(currentPage - minPage);
      const endReservePage = maxPage - (reserveBaseCount - 1);
      const endReserveBoundaryPage = endReservePage - 1;
      let ellipsis = false;
      for (let i = currentPage + 1; i <= maxPage; i++) {
        if (labels.length < reserveCount
          || i >= endReservePage
          || (i === endReserveBoundaryPage && !ellipsis)) {
          labels.push('' + i);
        } else if (!ellipsis) {
          ellipsis = true;
          labels.push(_ellipsisLabel);
        }
      }
      return labels;
    })();

    return {
      leftLabels: leftLabels.reverse(),
      currentLabel: '' + currentPage,
      rightLabels: rightLabels,
      previousPage: currentPage <= minPage ? 0 : currentPage - 1,
      nextPage: currentPage >= maxPage ? 0 : currentPage + 1
    };
  }

  static getOffset(limit, page) {
    return (page - 1) * limit;
  }

  handleClick(e, targetPage) {
    this.props.onClick && this.props.onClick(e, FlatPagination.getOffset(this.props.limit, targetPage));
  }

  renderButton(position, targetPage, label) {
    const {
      total,
      currentPageLabelStyle,
      currentPageStyle,
      disabled,
      disableTouchRipple,
      hoverColor,
      otherPageLabelStyle,
      otherPageStyle,
      rippleColor
    } = this.props;

    const totalZero = total <= 0;

    if (position === 'previous' || position === 'next') {
      let icon;
      if (React.isValidElement(label)) {
        icon = React.cloneElement(label, {
          style: {..._styles.labelStyle, ...label.props.style}
        });
      }
      return (
        <FlatButton
          key={position}
          {...(icon ? {icon: icon} : {label: label})}
          primary={true}
          disabled={disabled || totalZero || targetPage <= 0}
          disableTouchRipple={disableTouchRipple}
          labelStyle={{..._styles.labelStyle, ...otherPageLabelStyle}}
          hoverColor={hoverColor}
          onClick={e => this.handleClick(e, targetPage)}
          rippleColor={rippleColor}
          style={{..._styles.buttonStyle, ...otherPageStyle}}
        />
      );
    } else if (position === 'left' || position === 'right') {
      const isEllipsis = label === _ellipsisLabel;
      return (
        <FlatButton
          key={isEllipsis ? position + 'Ellipsis' : label}
          label={label}
          primary={true}
          disabled={disabled || totalZero || isEllipsis}
          disableTouchRipple={disableTouchRipple}
          labelStyle={{...(isEllipsis ? _styles.ellipsisLabelStyle : _styles.labelStyle), ...otherPageLabelStyle}}
          hoverColor={hoverColor}
          onClick={e => this.handleClick(e, targetPage)}
          rippleColor={rippleColor}
          style={{..._styles.buttonStyle, ...otherPageStyle}}
        />
      );
    } else {
      return (
        <FlatButton
          key={label}
          label={label}
          secondary={true}
          disabled={disabled || totalZero}
          disableTouchRipple={true}
          labelStyle={{..._styles.labelStyle, ...currentPageLabelStyle}}
          hoverColor={hoverColor}
          style={{..._styles.buttonStyle, ...currentPageStyle}}
        />
      );
    }
  }

  render() {
    const {
      offset,
      limit,
      total,
      className,
      nextPageLabel,
      previousPageLabel,
      reduced,
      style
    } = this.props;

    const {
      leftLabels,
      currentLabel,
      rightLabels,
      previousPage,
      nextPage
    } = FlatPagination.createLabel(offset, limit, total, reduced);

    const targetPage = (label) => parseInt(label, 10);

    return (
      <div className={'material-ui-flat-pagination' + (className ? ' ' + className : '')} style={style}>
        {this.renderButton('previous', previousPage, previousPageLabel)}
        {leftLabels.map(label => this.renderButton('left', targetPage(label), label))}
        {this.renderButton('current', targetPage(currentLabel), currentLabel)}
        {rightLabels.map(label => this.renderButton('right', targetPage(label), label))}
        {this.renderButton('next', nextPage, nextPageLabel)}
      </div>
    );
  }

}

export default FlatPagination;