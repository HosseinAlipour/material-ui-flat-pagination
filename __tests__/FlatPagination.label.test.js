'use strict';

import React from 'react';
import {mount} from 'enzyme';
import {mountToJson} from 'enzyme-to-json';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import FlatPagination from '../src/FlatPagination';
import NavigationArrowBack from 'material-ui/svg-icons/navigation/arrow-back';
import NavigationArrowForward from 'material-ui/svg-icons/navigation/arrow-forward';

describe('Pagination Label', () => {
  const wrapper = (commonProps, otherProps) => {
    const props = Object.assign({}, commonProps, otherProps);
    return mount(
      <MuiThemeProvider muiTheme={getMuiTheme()}>
        <FlatPagination {...props}/>
      </MuiThemeProvider>
    ).find('.material-ui-flat-pagination');
  };

  // ********************************************************************************
  describe('nextPageLabel', () => {
    const props = {offset: 0, limit: 10, total: 10, nextPageLabel: <NavigationArrowForward/>};

    it('React element (NavigationArrowForward)', () => {
      expect(wrapper(props).find('button').last().find('svg')).toHaveLength(1);
    });

    it('snapshot', () => {
      const target = mountToJson(wrapper(props));
      expect(target).toMatchSnapshot();
    });
  });

  // ********************************************************************************
  describe('previousPageLabel', () => {
    const props = {offset: 0, limit: 10, total: 10, previousPageLabel: <NavigationArrowBack/>};

    it('React element (NavigationArrowBack)', () => {
      expect(wrapper(props).find('button').first().find('svg')).toHaveLength(1);
    });

    it('snapshot', () => {
      const target = mountToJson(wrapper(props));
      expect(target).toMatchSnapshot();
    });
  });

});
