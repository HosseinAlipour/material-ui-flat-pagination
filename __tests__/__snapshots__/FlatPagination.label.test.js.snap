// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Label nextPageLabel snapshot 1`] = `
<div
  className="material-ui-flat-pagination"
>
  <FlatButton
    disableTouchRipple={false}
    disabled={true}
    fullWidth={false}
    key="previous"
    label="<"
    labelPosition="after"
    labelStyle={
      Object {
        "padding": "0 12px",
      }
    }
    onClick={[Function]}
    onKeyboardFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onTouchStart={[Function]}
    primary={true}
    secondary={false}
    style={
      Object {
        "minWidth": "none",
      }
    }
  >
    <EnhancedButton
      containerElement="button"
      disableTouchRipple={false}
      disabled={true}
      focusRippleColor="#999999"
      focusRippleOpacity={0.3}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "borderRadius": 2,
          "color": "rgba(0, 0, 0, 0.3)",
          "height": 36,
          "lineHeight": "36px",
          "margin": 0,
          "minWidth": "none",
          "overflow": "hidden",
          "padding": 0,
          "textAlign": "center",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
        }
      }
      tabIndex={0}
      touchRippleColor="#999999"
      touchRippleOpacity={0.3}
      type="button"
    >
      <button
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "border": 10,
            "borderRadius": 2,
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.3)",
            "cursor": "default",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": "inherit",
            "fontWeight": "inherit",
            "height": 36,
            "lineHeight": "36px",
            "margin": 0,
            "minWidth": "none",
            "muiPrepared": true,
            "outline": "none",
            "overflow": "hidden",
            "padding": 0,
            "position": "relative",
            "textAlign": "center",
            "textDecoration": "none",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "verticalAlign": null,
            "zIndex": 1,
          }
        }
        tabIndex={-1}
        type="button"
      >
        <FlatButtonLabel
          key="labelElement"
          label="<"
          style={
            Object {
              "fontSize": 14,
              "fontWeight": 500,
              "letterSpacing": 0,
              "padding": "0 12px",
              "textTransform": "uppercase",
            }
          }
        >
          <span
            style={
              Object {
                "fontSize": 14,
                "fontWeight": 500,
                "letterSpacing": 0,
                "muiPrepared": true,
                "padding": "0 12px",
                "paddingLeft": 16,
                "paddingRight": 16,
                "position": "relative",
                "textTransform": "uppercase",
                "verticalAlign": "middle",
              }
            }
          >
            &lt;
          </span>
        </FlatButtonLabel>
      </button>
    </EnhancedButton>
  </FlatButton>
  <FlatButton
    disableTouchRipple={true}
    disabled={false}
    fullWidth={false}
    key="1"
    label="1"
    labelPosition="after"
    labelStyle={
      Object {
        "padding": "0 12px",
      }
    }
    onKeyboardFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onTouchStart={[Function]}
    primary={false}
    secondary={true}
    style={
      Object {
        "minWidth": "none",
      }
    }
  >
    <EnhancedButton
      containerElement="button"
      disableTouchRipple={true}
      disabled={false}
      focusRippleColor="#999999"
      focusRippleOpacity={0.3}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "borderRadius": 2,
          "color": "#ff4081",
          "height": 36,
          "lineHeight": "36px",
          "margin": 0,
          "minWidth": "none",
          "overflow": "hidden",
          "padding": 0,
          "textAlign": "center",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
        }
      }
      tabIndex={0}
      touchRippleColor="#999999"
      touchRippleOpacity={0.3}
      type="button"
    >
      <button
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "border": 10,
            "borderRadius": 2,
            "boxSizing": "border-box",
            "color": "#ff4081",
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": "inherit",
            "fontWeight": "inherit",
            "height": 36,
            "lineHeight": "36px",
            "margin": 0,
            "minWidth": "none",
            "muiPrepared": true,
            "outline": "none",
            "overflow": "hidden",
            "padding": 0,
            "position": "relative",
            "textAlign": "center",
            "textDecoration": "none",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "verticalAlign": null,
            "zIndex": 1,
          }
        }
        tabIndex={0}
        type="button"
      >
        <FlatButtonLabel
          key="labelElement"
          label="1"
          style={
            Object {
              "fontSize": 14,
              "fontWeight": 500,
              "letterSpacing": 0,
              "padding": "0 12px",
              "textTransform": "uppercase",
            }
          }
        >
          <span
            style={
              Object {
                "fontSize": 14,
                "fontWeight": 500,
                "letterSpacing": 0,
                "muiPrepared": true,
                "padding": "0 12px",
                "paddingLeft": 16,
                "paddingRight": 16,
                "position": "relative",
                "textTransform": "uppercase",
                "verticalAlign": "middle",
              }
            }
          >
            1
          </span>
        </FlatButtonLabel>
      </button>
    </EnhancedButton>
  </FlatButton>
  <FlatButton
    disableTouchRipple={false}
    disabled={true}
    fullWidth={false}
    icon={
      <NavigationArrowForward
        style={
          Object {
            "padding": "0 12px",
          }
        }
      />
    }
    key="next"
    labelPosition="after"
    labelStyle={
      Object {
        "padding": "0 12px",
      }
    }
    onClick={[Function]}
    onKeyboardFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onTouchStart={[Function]}
    primary={true}
    secondary={false}
    style={
      Object {
        "minWidth": "none",
      }
    }
  >
    <EnhancedButton
      containerElement="button"
      disableTouchRipple={false}
      disabled={true}
      focusRippleColor="#999999"
      focusRippleOpacity={0.3}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "borderRadius": 2,
          "color": "rgba(0, 0, 0, 0.3)",
          "height": 36,
          "lineHeight": "36px",
          "margin": 0,
          "minWidth": "none",
          "overflow": "hidden",
          "padding": 0,
          "textAlign": "center",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
        }
      }
      tabIndex={0}
      touchRippleColor="#999999"
      touchRippleOpacity={0.3}
      type="button"
    >
      <button
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "border": 10,
            "borderRadius": 2,
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.3)",
            "cursor": "default",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": "inherit",
            "fontWeight": "inherit",
            "height": 36,
            "lineHeight": "36px",
            "margin": 0,
            "minWidth": "none",
            "muiPrepared": true,
            "outline": "none",
            "overflow": "hidden",
            "padding": 0,
            "position": "relative",
            "textAlign": "center",
            "textDecoration": "none",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "verticalAlign": null,
            "zIndex": 1,
          }
        }
        tabIndex={-1}
        type="button"
      >
        <NavigationArrowForward
          color="rgba(0, 0, 0, 0.3)"
          key="iconCloned"
          style={
            Object {
              "marginLeft": 0,
              "marginRight": 0,
              "padding": "0 12px",
              "verticalAlign": "middle",
            }
          }
        >
          <SvgIcon
            color="rgba(0, 0, 0, 0.3)"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "marginLeft": 0,
                "marginRight": 0,
                "padding": "0 12px",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 24 24"
          >
            <svg
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                  "display": "inline-block",
                  "fill": "rgba(0, 0, 0, 0.3)",
                  "height": 24,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "muiPrepared": true,
                  "padding": "0 12px",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "userSelect": "none",
                  "verticalAlign": "middle",
                  "width": 24,
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
              />
            </svg>
          </SvgIcon>
        </NavigationArrowForward>
      </button>
    </EnhancedButton>
  </FlatButton>
</div>
`;

exports[`Pagination Label previousPageLabel snapshot 1`] = `
<div
  className="material-ui-flat-pagination"
>
  <FlatButton
    disableTouchRipple={false}
    disabled={true}
    fullWidth={false}
    icon={
      <NavigationArrowBack
        style={
          Object {
            "padding": "0 12px",
          }
        }
      />
    }
    key="previous"
    labelPosition="after"
    labelStyle={
      Object {
        "padding": "0 12px",
      }
    }
    onClick={[Function]}
    onKeyboardFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onTouchStart={[Function]}
    primary={true}
    secondary={false}
    style={
      Object {
        "minWidth": "none",
      }
    }
  >
    <EnhancedButton
      containerElement="button"
      disableTouchRipple={false}
      disabled={true}
      focusRippleColor="#999999"
      focusRippleOpacity={0.3}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "borderRadius": 2,
          "color": "rgba(0, 0, 0, 0.3)",
          "height": 36,
          "lineHeight": "36px",
          "margin": 0,
          "minWidth": "none",
          "overflow": "hidden",
          "padding": 0,
          "textAlign": "center",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
        }
      }
      tabIndex={0}
      touchRippleColor="#999999"
      touchRippleOpacity={0.3}
      type="button"
    >
      <button
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "border": 10,
            "borderRadius": 2,
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.3)",
            "cursor": "default",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": "inherit",
            "fontWeight": "inherit",
            "height": 36,
            "lineHeight": "36px",
            "margin": 0,
            "minWidth": "none",
            "muiPrepared": true,
            "outline": "none",
            "overflow": "hidden",
            "padding": 0,
            "position": "relative",
            "textAlign": "center",
            "textDecoration": "none",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "verticalAlign": null,
            "zIndex": 1,
          }
        }
        tabIndex={-1}
        type="button"
      >
        <NavigationArrowBack
          color="rgba(0, 0, 0, 0.3)"
          key="iconCloned"
          style={
            Object {
              "marginLeft": 0,
              "marginRight": 0,
              "padding": "0 12px",
              "verticalAlign": "middle",
            }
          }
        >
          <SvgIcon
            color="rgba(0, 0, 0, 0.3)"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "marginLeft": 0,
                "marginRight": 0,
                "padding": "0 12px",
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 24 24"
          >
            <svg
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                  "display": "inline-block",
                  "fill": "rgba(0, 0, 0, 0.3)",
                  "height": 24,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "muiPrepared": true,
                  "padding": "0 12px",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "userSelect": "none",
                  "verticalAlign": "middle",
                  "width": 24,
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
              />
            </svg>
          </SvgIcon>
        </NavigationArrowBack>
      </button>
    </EnhancedButton>
  </FlatButton>
  <FlatButton
    disableTouchRipple={true}
    disabled={false}
    fullWidth={false}
    key="1"
    label="1"
    labelPosition="after"
    labelStyle={
      Object {
        "padding": "0 12px",
      }
    }
    onKeyboardFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onTouchStart={[Function]}
    primary={false}
    secondary={true}
    style={
      Object {
        "minWidth": "none",
      }
    }
  >
    <EnhancedButton
      containerElement="button"
      disableTouchRipple={true}
      disabled={false}
      focusRippleColor="#999999"
      focusRippleOpacity={0.3}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "borderRadius": 2,
          "color": "#ff4081",
          "height": 36,
          "lineHeight": "36px",
          "margin": 0,
          "minWidth": "none",
          "overflow": "hidden",
          "padding": 0,
          "textAlign": "center",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
        }
      }
      tabIndex={0}
      touchRippleColor="#999999"
      touchRippleOpacity={0.3}
      type="button"
    >
      <button
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "border": 10,
            "borderRadius": 2,
            "boxSizing": "border-box",
            "color": "#ff4081",
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": "inherit",
            "fontWeight": "inherit",
            "height": 36,
            "lineHeight": "36px",
            "margin": 0,
            "minWidth": "none",
            "muiPrepared": true,
            "outline": "none",
            "overflow": "hidden",
            "padding": 0,
            "position": "relative",
            "textAlign": "center",
            "textDecoration": "none",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "verticalAlign": null,
            "zIndex": 1,
          }
        }
        tabIndex={0}
        type="button"
      >
        <FlatButtonLabel
          key="labelElement"
          label="1"
          style={
            Object {
              "fontSize": 14,
              "fontWeight": 500,
              "letterSpacing": 0,
              "padding": "0 12px",
              "textTransform": "uppercase",
            }
          }
        >
          <span
            style={
              Object {
                "fontSize": 14,
                "fontWeight": 500,
                "letterSpacing": 0,
                "muiPrepared": true,
                "padding": "0 12px",
                "paddingLeft": 16,
                "paddingRight": 16,
                "position": "relative",
                "textTransform": "uppercase",
                "verticalAlign": "middle",
              }
            }
          >
            1
          </span>
        </FlatButtonLabel>
      </button>
    </EnhancedButton>
  </FlatButton>
  <FlatButton
    disableTouchRipple={false}
    disabled={true}
    fullWidth={false}
    key="next"
    label=">"
    labelPosition="after"
    labelStyle={
      Object {
        "padding": "0 12px",
      }
    }
    onClick={[Function]}
    onKeyboardFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onTouchStart={[Function]}
    primary={true}
    secondary={false}
    style={
      Object {
        "minWidth": "none",
      }
    }
  >
    <EnhancedButton
      containerElement="button"
      disableTouchRipple={false}
      disabled={true}
      focusRippleColor="#999999"
      focusRippleOpacity={0.3}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "borderRadius": 2,
          "color": "rgba(0, 0, 0, 0.3)",
          "height": 36,
          "lineHeight": "36px",
          "margin": 0,
          "minWidth": "none",
          "overflow": "hidden",
          "padding": 0,
          "textAlign": "center",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
        }
      }
      tabIndex={0}
      touchRippleColor="#999999"
      touchRippleOpacity={0.3}
      type="button"
    >
      <button
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "border": 10,
            "borderRadius": 2,
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.3)",
            "cursor": "default",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": "inherit",
            "fontWeight": "inherit",
            "height": 36,
            "lineHeight": "36px",
            "margin": 0,
            "minWidth": "none",
            "muiPrepared": true,
            "outline": "none",
            "overflow": "hidden",
            "padding": 0,
            "position": "relative",
            "textAlign": "center",
            "textDecoration": "none",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "verticalAlign": null,
            "zIndex": 1,
          }
        }
        tabIndex={-1}
        type="button"
      >
        <FlatButtonLabel
          key="labelElement"
          label=">"
          style={
            Object {
              "fontSize": 14,
              "fontWeight": 500,
              "letterSpacing": 0,
              "padding": "0 12px",
              "textTransform": "uppercase",
            }
          }
        >
          <span
            style={
              Object {
                "fontSize": 14,
                "fontWeight": 500,
                "letterSpacing": 0,
                "muiPrepared": true,
                "padding": "0 12px",
                "paddingLeft": 16,
                "paddingRight": 16,
                "position": "relative",
                "textTransform": "uppercase",
                "verticalAlign": "middle",
              }
            }
          >
            &gt;
          </span>
        </FlatButtonLabel>
      </button>
    </EnhancedButton>
  </FlatButton>
</div>
`;
